### ISSUE TITLE
Split Docker Engine into a distinct role

### ISSUE THREAD

**Issue by kinghuang** [atosatto/ansible-dockerswarm#5]
I'm wondering if you'd consider splitting off the Docker Engine tasks into its own role? I have some hosts that just need to be set up as engines, without swarm mode.

Your tasks for Docker Engine are nicely written with multi-OS support. It'd be useful if it existed as a distinct role that the docker-swarm role depended on rather than being bundled within the docker-swarm role.

**Comment by atosatto** [comment]
This is a very good suggestion I also had from @rothgar on twitter.
I don't like very much the idea of splitting the role because I would like a "docker-engine-only" role to do lot more of what this role is currently doing (eg. customisation of the docker-engine configuration).  
But I believe that you're right. So I'm gonna add to this role some switches to "install the docker-engine without swarm" or "swarm without the docker-engine".

**Comment by kinghuang** [comment]
Yes, some sort of switch to enable ot disable swarm mode would work, too. Thanks for considering this!

**Comment by atosatto** [comment]
I've just released `v1.2.2` implementing you're suggestion.
You should now be able to disable `swarm-mode`with something like the following plugin 

```
---

- hosts: all
  roles:
    - { role: atosatto.docker-swarm,
        skip_swarm: True }
```
